tools/mock_node/benches/sync.rs

```diff
@@ -0,0 +1,127 @@
+#[macro_use]
+extern crate criterion;
+
+use actix::System;
+use anyhow::anyhow;
+use criterion::Criterion;
+use flate2::read::GzDecoder;
+use mock_node::setup::{setup_mock_node, MockNetworkMode};
+use mock_node::GetChainTargetBlockHeight;
+use near_actix_test_utils::{block_on_interruptible, setup_actix};
+use near_chain_configs::GenesisValidationMode;
+use near_client::GetBlock;
+use near_crypto::{InMemorySigner, KeyType};
+use near_primitives::types::BlockHeight;
+use std::fs::File;
+use std::path::Path;
+use std::time::{Duration, Instant};
+use tar::Archive;
+
+struct Sys {
+    sys: Option<actix_rt::SystemRunner>,
+    servers: Vec<(&'static str, actix_web::dev::Server)>,
+}
+
+impl Drop for Sys {
+    fn drop(&mut self) {
+        // TODO: we only have to do this because shutdown is not well handled right now. Ideally we would not have
+        // to tear down the whole system, and could just stop the client actor/view client actors each time.
+        let system = System::current();
+        let sys = self.sys.take().unwrap();
+
+        sys.block_on(async move {
+            futures::future::join_all(self.servers.iter().map(|(_name, server)| async move {
+                server.stop(true).await;
+            }))
+            .await;
+        });
+        system.stop();
+        sys.run().unwrap();
+        near_store::db::RocksDB::block_until_all_instances_are_dropped();
+    }
+}
+
+fn extract_home(home_archive: &str) -> anyhow::Result<&Path> {
+    let len = home_archive.len();
+    if len <= 7 || &home_archive[len - 7..] != ".tar.gz" {
+        return Err(anyhow!("{} doesn't end with .tar.gz", home_archive));
+    }
+    let extracted = Path::new(&home_archive[..len - 7]);
+    if extracted.exists() {
+        return Ok(extracted);
+    }
+
+    let tar_gz = File::open(home_archive)?;
+    let tar = GzDecoder::new(tar_gz);
+    let mut archive = Archive::new(tar);
+    archive.unpack(extracted)?;
```

Comment:
so we're extracting it 'into' our code/github directory? 

that's not great (as it can pollute it for some people).
Maybe we should extract it to temp directory instead?

Reply:
yeah good point. updated to extract to /tmp/near_mock_node_sync_{testname}

Reply:
also fixed a problem where it was extracting all archives every time, even if you specify a particular test on the cmdline